<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CRSPlannerConfig</class>
 <widget class="QWidget" name="CRSPlannerConfig">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>260</width>
    <height>591</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CRS Planning Config</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="margin">
    <number>1</number>
   </property>
   <item row="0" column="0">
    <widget class="QTabWidget" name="config_panels">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="tabPosition">
      <enum>QTabWidget::South</enum>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="TopologyTab">
      <attribute name="title">
       <string>Space</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Differential Order of the Joint-Space:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="order_selection">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <item>
          <property name="text">
           <string>0th-Order Space (positions only)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>1st-Order Space (position + velocity)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>2nd-Order Space (pos. + vel. + acc.)</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Interpolation Method:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="interp_selection">
         <item>
          <property name="text">
           <string>Linear Interpolation</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Cubic Interpolation</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Quintic Interpolation</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Sustained Velocity Pulses</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Sustained Acceleration Pulses</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Minimum Travel Interval (seconds):</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="min_interval_spinbox">
         <property name="minimum">
          <double>0.010000000000000</double>
         </property>
         <property name="maximum">
          <double>20.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
         <property name="value">
          <double>0.100000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Maximum Travel Interval (seconds):</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="max_interval_spinbox">
         <property name="minimum">
          <double>0.100000000000000</double>
         </property>
         <property name="maximum">
          <double>100.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>1.000000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_13">
         <property name="text">
          <string>Starting Joint Configuration:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBox">
         <property name="text">
          <string>Current Configuration</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBox_2">
         <property name="text">
          <string>Joint Configuration below:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit">
         <property name="text">
          <string>(0; 0; 0; 0; 0; 0; 0)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_14">
         <property name="text">
          <string>Target Joint Configuration:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBox_3">
         <property name="text">
          <string>Current Target End-effector Pose (IK)</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBox_4">
         <property name="text">
          <string>Joint Configuration below:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_2">
         <property name="text">
          <string>(0; 0; 0; 0; 0; 0; 0)</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="PlannerTab">
      <attribute name="title">
       <string>Planner</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Motion Planning Method:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="planning_algo_selection">
         <item>
          <property name="text">
           <string>Rapidly-Exploring Random Tree (RRT)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Rapidly-Exploring Random Tree Star (RRT*)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Rapidly-Exploring Random Graph (RRG)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Probabilistic Road-Map (PRM)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Probabilistic Road-Map Star (PRM*)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Flexible Anytime-Dynamic PRM (FADPRM)</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Directionality of the Planner:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="direction_selection">
         <item>
          <property name="text">
           <string>Uni-directional</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Bi-directional</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Motion-Graph Storage Policy:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="graph_storage_selection">
         <item>
          <property name="text">
           <string>Adjacency-List Graph</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Adjacency-List / DVP-Tree Overlay</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Nearest-Neighbor Method:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="KNN_method_selection">
         <item>
          <property name="text">
           <string>Linear Search</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Binary Breadth-first DVP-Tree</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Quaternary Breadth-first DVP-Tree</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Binary Cache-Oblivious DVP-tree</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Quaternary Cache-Oblivious DVP-tree</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>Maximum Number of Nodes:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="maxvertices_spinbox">
         <property name="minimum">
          <number>100</number>
         </property>
         <property name="maximum">
          <number>20000</number>
         </property>
         <property name="singleStep">
          <number>100</number>
         </property>
         <property name="value">
          <number>2000</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>Progress-Report Interval (node-count):</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="progress_interval_spinbox">
         <property name="minimum">
          <number>100</number>
         </property>
         <property name="maximum">
          <number>10000</number>
         </property>
         <property name="singleStep">
          <number>100</number>
         </property>
         <property name="value">
          <number>500</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>Maximum Number of Solutions:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="maxsolutions_spinbox">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>200</number>
         </property>
         <property name="singleStep">
          <number>5</number>
         </property>
         <property name="value">
          <number>50</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_12">
         <property name="text">
          <string>Progress to be reported:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="check_print_timing">
         <property name="text">
          <string>Timing (printed to terminal)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="check_print_counter">
         <property name="text">
          <string>Node counter (printed to terminal)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="check_print_graph">
         <property name="text">
          <string>Motion-graph (end-effector trace)</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="check_print_best">
         <property name="text">
          <string>Best Solution (end-effector trace)</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="check_print_allsol">
         <property name="text">
          <string>All Solutions (end-effector traces)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton">
         <property name="text">
          <string>Execute Motion Planning...</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="DisplayTab">
      <attribute name="title">
       <string>Display</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QCheckBox" name="checkBox_5">
         <property name="text">
          <string>Show Robot Geometry</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBox_6">
         <property name="text">
          <string>Show Robot Kinematic Model</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBox_7">
         <property name="text">
          <string>Show Target Geometry</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBox_8">
         <property name="text">
          <string>Show Environment</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBox_9">
         <property name="text">
          <string>Show Proximity Query Results</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_15">
         <property name="text">
          <string>Motion-Planning Results:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBox_10">
         <property name="text">
          <string>Show Motion-Graph</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBox_11">
         <property name="text">
          <string>Show Solution(s)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_2">
         <property name="text">
          <string>Animate Best Solution</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="InteractTab">
      <attribute name="title">
       <string>Interact</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QCheckBox" name="checkBox_12">
         <property name="text">
          <string>Enable Inverse Kinematics</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="Line" name="line_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_16">
         <property name="text">
          <string>Joint Positions (track + 6 joints):</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="track_pos">
         <property name="maximum">
          <number>3000</number>
         </property>
         <property name="singleStep">
          <number>10</number>
         </property>
         <property name="pageStep">
          <number>10</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="joint1_pos">
         <property name="minimum">
          <number>-3054</number>
         </property>
         <property name="maximum">
          <number>3054</number>
         </property>
         <property name="singleStep">
          <number>10</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="joint2_pos">
         <property name="minimum">
          <number>-1570</number>
         </property>
         <property name="maximum">
          <number>1570</number>
         </property>
         <property name="singleStep">
          <number>10</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="joint3_pos">
         <property name="minimum">
          <number>-1919</number>
         </property>
         <property name="maximum">
          <number>1919</number>
         </property>
         <property name="singleStep">
          <number>10</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="joint4_pos">
         <property name="minimum">
          <number>-3141</number>
         </property>
         <property name="maximum">
          <number>3141</number>
         </property>
         <property name="singleStep">
          <number>10</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="joint5_pos">
         <property name="minimum">
          <number>-1832</number>
         </property>
         <property name="singleStep">
          <number>10</number>
         </property>
         <property name="value">
          <number>0</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="joint6_pos">
         <property name="minimum">
          <number>-3141</number>
         </property>
         <property name="maximum">
          <number>3141</number>
         </property>
         <property name="singleStep">
          <number>10</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="Line" name="line_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_17">
         <property name="text">
          <string>Target Pose:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_18">
         <property name="text">
          <string>  Position (x, y, z):</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="target_x">
         <property name="minimum">
          <number>-4000</number>
         </property>
         <property name="maximum">
          <number>2000</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="target_y">
         <property name="minimum">
          <number>-4000</number>
         </property>
         <property name="singleStep">
          <number>50</number>
         </property>
         <property name="pageStep">
          <number>50</number>
         </property>
         <property name="value">
          <number>0</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="target_z">
         <property name="maximum">
          <number>3000</number>
         </property>
         <property name="value">
          <number>1500</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_19">
         <property name="text">
          <string>  Orientation (yaw, pitch and roll):</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="target_yaw">
         <property name="minimum">
          <number>-180</number>
         </property>
         <property name="maximum">
          <number>180</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="target_pitch">
         <property name="minimum">
          <number>-180</number>
         </property>
         <property name="maximum">
          <number>180</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="target_roll">
         <property name="minimum">
          <number>-180</number>
         </property>
         <property name="maximum">
          <number>180</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
